File: php/SwaggerClient-php/lib/Model/V3SearchParameters.php
<?php
/**
 * V3SearchParameters
 *
 * PHP version 5
 *
 * @category Class
 * @package  Swagger\Client
 * @author   Swagger Codegen team
 * @link     https://github.com/swagger-api/swagger-codegen
 */

/**
 * PTV Timetable API - Version 3
 *
 * The PTV Timetable API provides direct access to Public Transport Victoria's public transport timetable data.    The API returns scheduled timetable, route and stop data for all metropolitan and regional train, tram and bus services in Victoria, including Night Network(Night Train and Night Tram data are included in metropolitan train and tram services data, respectively, whereas Night Bus is a separate route type).    The API also returns real-time data for metropolitan train, tram and bus services (where this data is made available to PTV), as well as disruption information, stop facility information, and access to myki ticket outlet data.    This Swagger is for Version 3 of the PTV Timetable API. By using this documentation you agree to comply with the licence and terms of service.    Train timetable data is updated daily, while the remaining data is updated weekly, taking into account any planned timetable changes (for example, due to holidays or planned disruptions). The PTV timetable API is the same API used by PTV for its apps. To access the most up to date data PTV has (including real-time data) you must use the API dynamically.    You can access the PTV Timetable API through a HTTP or HTTPS interface, as follows:        base URL / version number / API name / query string  The base URL is either:    *  http://timetableapi.ptv.vic.gov.au  or    *  https://timetableapi.ptv.vic.gov.au    The Swagger JSON file is available at http://timetableapi.ptv.vic.gov.au/swagger/docs/v3    Frequently asked questions are available on the PTV website at http://ptv.vic.gov.au/apifaq    Links to the following information are also provided on the PTV website at http://ptv.vic.gov.au/ptv-timetable-api/  * How to register for an API key and calculate a signature  * PTV Timetable API V2 to V3 Migration Guide  * PTV Timetable API Data Quality Statement    All information about how to use the API is in this documentation. PTV cannot provide technical support for the API.
 *
 * OpenAPI spec version: v3
 * 
 * Generated by: https://github.com/swagger-api/swagger-codegen.git
 * Swagger Codegen version: 3.0.52
 */
/**
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen
 * Do not edit the class manually.
 */

namespace Swagger\Client\Model;

use \ArrayAccess;
use \Swagger\Client\ObjectSerializer;

/**
 * V3SearchParameters Class Doc Comment
 *
 * @category Class
 * @package  Swagger\Client
 * @author   Swagger Codegen team
 * @link     https://github.com/swagger-api/swagger-codegen
 */
class V3SearchParameters implements ModelInterface, ArrayAccess
{
    const DISCRIMINATOR = null;

    /**
      * The original name of the model.
      *
      * @var string
      */
    protected static $swaggerModelName = 'V3.SearchParameters';

    /**
      * Array of property to type mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $swaggerTypes = [
        'route_types' => 'int[]',
        'latitude' => 'float',
        'longitude' => 'float',
        'max_distance' => 'float',
        'include_addresses' => 'bool',
        'include_outlets' => 'bool',
        'match_stop_by_suburb' => 'bool',
        'match_route_by_suburb' => 'bool',
        'match_stop_by_gtfs_stop_id' => 'bool'
    ];

    /**
      * Array of property to format mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $swaggerFormats = [
        'route_types' => 'int32',
        'latitude' => 'float',
        'longitude' => 'float',
        'max_distance' => 'float',
        'include_addresses' => null,
        'include_outlets' => null,
        'match_stop_by_suburb' => null,
        'match_route_by_suburb' => null,
        'match_stop_by_gtfs_stop_id' => null
    ];

    /**
     * Array of property to type mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function swaggerTypes()
    {
        return self::$swaggerTypes;
    }

    /**
     * Array of property to format mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function swaggerFormats()
    {
        return self::$swaggerFormats;
    }

    /**
     * Array of attributes where the key is the local name,
     * and the value is the original name
     *
     * @var string[]
     */
    protected static $attributeMap = [
        'route_types' => 'route_types',
        'latitude' => 'latitude',
        'longitude' => 'longitude',
        'max_distance' => 'max_distance',
        'include_addresses' => 'include_addresses',
        'include_outlets' => 'include_outlets',
        'match_stop_by_suburb' => 'match_stop_by_suburb',
        'match_route_by_suburb' => 'match_route_by_suburb',
        'match_stop_by_gtfs_stop_id' => 'match_stop_by_gtfs_stop_id'
    ];

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @var string[]
     */
    protected static $setters = [
        'route_types' => 'setRouteTypes',
        'latitude' => 'setLatitude',
        'longitude' => 'setLongitude',
        'max_distance' => 'setMaxDistance',
        'include_addresses' => 'setIncludeAddresses',
        'include_outlets' => 'setIncludeOutlets',
        'match_stop_by_suburb' => 'setMatchStopBySuburb',
        'match_route_by_suburb' => 'setMatchRouteBySuburb',
        'match_stop_by_gtfs_stop_id' => 'setMatchStopByGtfsStopId'
    ];

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @var string[]
     */
    protected static $getters = [
        'route_types' => 'getRouteTypes',
        'latitude' => 'getLatitude',
        'longitude' => 'getLongitude',
        'max_distance' => 'getMaxDistance',
        'include_addresses' => 'getIncludeAddresses',
        'include_outlets' => 'getIncludeOutlets',
        'match_stop_by_suburb' => 'getMatchStopBySuburb',
        'match_route_by_suburb' => 'getMatchRouteBySuburb',
        'match_stop_by_gtfs_stop_id' => 'getMatchStopByGtfsStopId'
    ];

    /**
     * Array of attributes where the key is the local name,
     * and the value is the original name
     *
     * @return array
     */
    public static function attributeMap()
    {
        return self::$attributeMap;
    }

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @return array
     */
    public static function setters()
    {
        return self::$setters;
    }

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @return array
     */
    public static function getters()
    {
        return self::$getters;
    }

    /**
     * The original name of the model.
     *
     * @return string
     */
    public function getModelName()
    {
        return self::$swaggerModelName;
    }

    const ROUTE_TYPES__0 = '0';
    const ROUTE_TYPES__1 = '1';
    const ROUTE_TYPES__2 = '2';
    const ROUTE_TYPES__3 = '3';
    const ROUTE_TYPES__4 = '4';

    /**
     * Gets allowable values of the enum
     *
     * @return string[]
     */
    public function getRouteTypesAllowableValues()
    {
        return [
            self::ROUTE_TYPES__0,
            self::ROUTE_TYPES__1,
            self::ROUTE_TYPES__2,
            self::ROUTE_TYPES__3,
            self::ROUTE_TYPES__4,
        ];
    }

    /**
     * Associative array for storing property values
     *
     * @var mixed[]
     */
    protected $container = [];

    /**
     * Constructor
     *
     * @param mixed[] $data Associated array of property values
     *                      initializing the model
     */
    public function __construct(array $data = null)
    {
        $this->container['route_types'] = isset($data['route_types']) ? $data['route_types'] : null;
        $this->container['latitude'] = isset($data['latitude']) ? $data['latitude'] : null;
        $this->container['longitude'] = isset($data['longitude']) ? $data['longitude'] : null;
        $this->container['max_distance'] = isset($data['max_distance']) ? $data['max_distance'] : null;
        $this->container['include_addresses'] = isset($data['include_addresses']) ? $data['include_addresses'] : null;
        $this->container['include_outlets'] = isset($data['include_outlets']) ? $data['include_outlets'] : null;
        $this->container['match_stop_by_suburb'] = isset($data['match_stop_by_suburb']) ? $data['match_stop_by_suburb'] : null;
        $this->container['match_route_by_suburb'] = isset($data['match_route_by_suburb']) ? $data['match_route_by_suburb'] : null;
        $this->container['match_stop_by_gtfs_stop_id'] = isset($data['match_stop_by_gtfs_stop_id']) ? $data['match_stop_by_gtfs_stop_id'] : null;
    }

    /**
     * Show all the invalid properties with reasons.
     *
     * @return array invalid properties with reasons
     */
    public function listInvalidProperties()
    {
        $invalidProperties = [];

        return $invalidProperties;
    }

    /**
     * Validate all the properties in the model
     * return true if all passed
     *
     * @return bool True if all properties are valid
     */
    public function valid()
    {
        return count($this->listInvalidProperties()) === 0;
    }


    /**
     * Gets route_types
     *
     * @return int[]
     */
    public function getRouteTypes()
    {
        return $this->container['route_types'];
    }

    /**
     * Sets route_types
     *
     * @param int[] $route_types Filter by route_type; values returned via RouteTypes API (note: stops and routes are ordered by route_types specified)
     *
     * @return $this
     */
    public function setRouteTypes($route_types)
    {
        $allowedValues = $this->getRouteTypesAllowableValues();
        if (!is_null($route_types) && array_diff($route_types, $allowedValues)) {
            throw new \InvalidArgumentException(
                sprintf(
                    "Invalid value for 'route_types', must be one of '%s'",
                    implode("', '", $allowedValues)
                )
            );
        }
        $this->container['route_types'] = $route_types;

        return $this;
    }

    /**
     * Gets latitude
     *
     * @return float
     */
    public function getLatitude()
    {
        return $this->container['latitude'];
    }

    /**
     * Sets latitude
     *
     * @param float $latitude Filter by geographic coordinate of latitude
     *
     * @return $this
     */
    public function setLatitude($latitude)
    {
        $this->container['latitude'] = $latitude;

        return $this;
    }

    /**
     * Gets longitude
     *
     * @return float
     */
    public function getLongitude()
    {
        return $this->container['longitude'];
    }

    /**
     * Sets longitude
     *
     * @param float $longitude Filter by geographic coordinate of longitude
     *
     * @return $this
     */
    public function setLongitude($longitude)
    {
        $this->container['longitude'] = $longitude;

        return $this;
    }

    /**
     * Gets max_distance
     *
     * @return float
     */
    public function getMaxDistance()
    {
        return $this->container['max_distance'];
    }

    /**
     * Sets max_distance
     *
     * @param float $max_distance Filter by maximum distance (in metres) from location specified via latitude and longitude parameters
     *
     * @return $this
     */
    public function setMaxDistance($max_distance)
    {
        $this->container['max_distance'] = $max_distance;

        return $this;
    }

    /**
     * Gets include_addresses
     *
     * @return bool
     */
    public function getIncludeAddresses()
    {
        return $this->container['include_addresses'];
    }

    /**
     * Sets include_addresses
     *
     * @param bool $include_addresses Placeholder for future development; currently unavailable
     *
     * @return $this
     */
    public function setIncludeAddresses($include_addresses)
    {
        $this->container['include_addresses'] = $include_addresses;

        return $this;
    }

    /**
     * Gets include_outlets
     *
     * @return bool
     */
    public function getIncludeOutlets()
    {
        return $this->container['include_outlets'];
    }

    /**
     * Sets include_outlets
     *
     * @param bool $include_outlets Indicates if outlets will be returned in response (default = true)
     *
     * @return $this
     */
    public function setIncludeOutlets($include_outlets)
    {
        $this->container['include_outlets'] = $include_outlets;

        return $this;
    }

    /**
     * Gets match_stop_by_suburb
     *
     * @return bool
     */
    public function getMatchStopBySuburb()
    {
        return $this->container['match_stop_by_suburb'];
    }

    /**
     * Sets match_stop_by_suburb
     *
     * @param bool $match_stop_by_suburb Indicates whether to find stops by suburbs in the search term (default = true)
     *
     * @return $this
     */
    public function setMatchStopBySuburb($match_stop_by_suburb)
    {
        $this->container['match_stop_by_suburb'] = $match_stop_by_suburb;

        return $this;
    }

    /**
     * Gets match_route_by_suburb
     *
     * @return bool
     */
    public function getMatchRouteBySuburb()
    {
        return $this->container['match_route_by_suburb'];
    }

    /**
     * Sets match_route_by_suburb
     *
     * @param bool $match_route_by_suburb Indicates whether to find routes by suburbs in the search term (default = true)
     *
     * @return $this
     */
    public function setMatchRouteBySuburb($match_route_by_suburb)
    {
        $this->container['match_route_by_suburb'] = $match_route_by_suburb;

        return $this;
    }

    /**
     * Gets match_stop_by_gtfs_stop_id
     *
     * @return bool
     */
    public function getMatchStopByGtfsStopId()
    {
        return $this->container['match_stop_by_gtfs_stop_id'];
    }

    /**
     * Sets match_stop_by_gtfs_stop_id
     *
     * @param bool $match_stop_by_gtfs_stop_id Indicates whether to search for stops according to a metlink stop ID (default = false)
     *
     * @return $this
     */
    public function setMatchStopByGtfsStopId($match_stop_by_gtfs_stop_id)
    {
        $this->container['match_stop_by_gtfs_stop_id'] = $match_stop_by_gtfs_stop_id;

        return $this;
    }
    /**
     * Returns true if offset exists. False otherwise.
     *
     * @param integer $offset Offset
     *
     * @return boolean
     */
    #[\ReturnTypeWillChange]
    public function offsetExists($offset)
    {
        return isset($this->container[$offset]);
    }

    /**
     * Gets offset.
     *
     * @param integer $offset Offset
     *
     * @return mixed
     */
    #[\ReturnTypeWillChange]
    public function offsetGet($offset)
    {
        return isset($this->container[$offset]) ? $this->container[$offset] : null;
    }

    /**
     * Sets value based on offset.
     *
     * @param integer $offset Offset
     * @param mixed   $value  Value to be set
     *
     * @return void
     */
    #[\ReturnTypeWillChange]
    public function offsetSet($offset, $value)
    {
        if (is_null($offset)) {
            $this->container[] = $value;
        } else {
            $this->container[$offset] = $value;
        }
    }

    /**
     * Unsets offset.
     *
     * @param integer $offset Offset
     *
     * @return void
     */
    #[\ReturnTypeWillChange]
    public function offsetUnset($offset)
    {
        unset($this->container[$offset]);
    }

    /**
     * Gets the string presentation of the object
     *
     * @return string
     */
    public function __toString()
    {
        if (defined('JSON_PRETTY_PRINT')) { // use JSON pretty print
            return json_encode(
                ObjectSerializer::sanitizeForSerialization($this),
                JSON_PRETTY_PRINT
            );
        }

        return json_encode(ObjectSerializer::sanitizeForSerialization($this));
    }
}

File: openapi.3.0.1/swagger-3/python/swagger_client/models/v3_search_parameters.py
# coding: utf-8

"""
    PTV Timetable API - Version 3

    The PTV Timetable API provides direct access to Public Transport Victoria's public transport timetable data.    The API returns scheduled timetable, route and stop data for all metropolitan and regional train, tram and bus services in Victoria, including Night Network(Night Train and Night Tram data are included in metropolitan train and tram services data, respectively, whereas Night Bus is a separate route type).    The API also returns real-time data for metropolitan train, tram and bus services (where this data is made available to PTV), as well as disruption information, stop facility information, and access to myki ticket outlet data.    This Swagger is for Version 3 of the PTV Timetable API. By using this documentation you agree to comply with the licence and terms of service.    Train timetable data is updated daily, while the remaining data is updated weekly, taking into account any planned timetable changes (for example, due to holidays or planned disruptions). The PTV timetable API is the same API used by PTV for its apps. To access the most up to date data PTV has (including real-time data) you must use the API dynamically.    You can access the PTV Timetable API through a HTTP or HTTPS interface, as follows:        base URL / version number / API name / query string  The base URL is either:    *  http://timetableapi.ptv.vic.gov.au  or    *  https://timetableapi.ptv.vic.gov.au    The Swagger JSON file is available at http://timetableapi.ptv.vic.gov.au/swagger/docs/v3    Frequently asked questions are available on the PTV website at http://ptv.vic.gov.au/apifaq    Links to the following information are also provided on the PTV website at http://ptv.vic.gov.au/ptv-timetable-api/  * How to register for an API key and calculate a signature  * PTV Timetable API V2 to V3 Migration Guide  * PTV Timetable API Data Quality Statement    All information about how to use the API is in this documentation. PTV cannot provide technical support for the API.    # noqa: E501

    OpenAPI spec version: v3
    
    Generated by: https://github.com/swagger-api/swagger-codegen.git
"""

import pprint
import re  # noqa: F401

import six

class V3SearchParameters(object):
    """NOTE: This class is auto generated by the swagger code generator program.

    Do not edit the class manually.
    """
    """
    Attributes:
      swagger_types (dict): The key is attribute name
                            and the value is attribute type.
      attribute_map (dict): The key is attribute name
                            and the value is json key in definition.
    """
    swagger_types = {
        'route_types': 'list[int]',
        'latitude': 'float',
        'longitude': 'float',
        'max_distance': 'float',
        'include_addresses': 'bool',
        'include_outlets': 'bool',
        'match_stop_by_suburb': 'bool',
        'match_route_by_suburb': 'bool',
        'match_stop_by_gtfs_stop_id': 'bool'
    }

    attribute_map = {
        'route_types': 'route_types',
        'latitude': 'latitude',
        'longitude': 'longitude',
        'max_distance': 'max_distance',
        'include_addresses': 'include_addresses',
        'include_outlets': 'include_outlets',
        'match_stop_by_suburb': 'match_stop_by_suburb',
        'match_route_by_suburb': 'match_route_by_suburb',
        'match_stop_by_gtfs_stop_id': 'match_stop_by_gtfs_stop_id'
    }

    def __init__(self, route_types=None, latitude=None, longitude=None, max_distance=None, include_addresses=None, include_outlets=None, match_stop_by_suburb=None, match_route_by_suburb=None, match_stop_by_gtfs_stop_id=None):  # noqa: E501
        """V3SearchParameters - a model defined in Swagger"""  # noqa: E501
        self._route_types = None
        self._latitude = None
        self._longitude = None
        self._max_distance = None
        self._include_addresses = None
        self._include_outlets = None
        self._match_stop_by_suburb = None
        self._match_route_by_suburb = None
        self._match_stop_by_gtfs_stop_id = None
        self.discriminator = None
        if route_types is not None:
            self.route_types = route_types
        if latitude is not None:
            self.latitude = latitude
        if longitude is not None:
            self.longitude = longitude
        if max_distance is not None:
            self.max_distance = max_distance
        if include_addresses is not None:
            self.include_addresses = include_addresses
        if include_outlets is not None:
            self.include_outlets = include_outlets
        if match_stop_by_suburb is not None:
            self.match_stop_by_suburb = match_stop_by_suburb
        if match_route_by_suburb is not None:
            self.match_route_by_suburb = match_route_by_suburb
        if match_stop_by_gtfs_stop_id is not None:
            self.match_stop_by_gtfs_stop_id = match_stop_by_gtfs_stop_id

    @property
    def route_types(self):
        """Gets the route_types of this V3SearchParameters.  # noqa: E501

        Filter by route_type; values returned via RouteTypes API (note: stops and routes are ordered by route_types specified)  # noqa: E501

        :return: The route_types of this V3SearchParameters.  # noqa: E501
        :rtype: list[int]
        """
        return self._route_types

    @route_types.setter
    def route_types(self, route_types):
        """Sets the route_types of this V3SearchParameters.

        Filter by route_type; values returned via RouteTypes API (note: stops and routes are ordered by route_types specified)  # noqa: E501

        :param route_types: The route_types of this V3SearchParameters.  # noqa: E501
        :type: list[int]
        """
        allowed_values = [0, 1, 2, 3, 4]  # noqa: E501
        if not set(route_types).issubset(set(allowed_values)):
            raise ValueError(
                "Invalid values for `route_types` [{0}], must be a subset of [{1}]"  # noqa: E501
                .format(", ".join(map(str, set(route_types) - set(allowed_values))),  # noqa: E501
                        ", ".join(map(str, allowed_values)))
            )

        self._route_types = route_types

    @property
    def latitude(self):
        """Gets the latitude of this V3SearchParameters.  # noqa: E501

        Filter by geographic coordinate of latitude  # noqa: E501

        :return: The latitude of this V3SearchParameters.  # noqa: E501
        :rtype: float
        """
        return self._latitude

    @latitude.setter
    def latitude(self, latitude):
        """Sets the latitude of this V3SearchParameters.

        Filter by geographic coordinate of latitude  # noqa: E501

        :param latitude: The latitude of this V3SearchParameters.  # noqa: E501
        :type: float
        """

        self._latitude = latitude

    @property
    def longitude(self):
        """Gets the longitude of this V3SearchParameters.  # noqa: E501

        Filter by geographic coordinate of longitude  # noqa: E501

        :return: The longitude of this V3SearchParameters.  # noqa: E501
        :rtype: float
        """
        return self._longitude

    @longitude.setter
    def longitude(self, longitude):
        """Sets the longitude of this V3SearchParameters.

        Filter by geographic coordinate of longitude  # noqa: E501

        :param longitude: The longitude of this V3SearchParameters.  # noqa: E501
        :type: float
        """

        self._longitude = longitude

    @property
    def max_distance(self):
        """Gets the max_distance of this V3SearchParameters.  # noqa: E501

        Filter by maximum distance (in metres) from location specified via latitude and longitude parameters  # noqa: E501

        :return: The max_distance of this V3SearchParameters.  # noqa: E501
        :rtype: float
        """
        return self._max_distance

    @max_distance.setter
    def max_distance(self, max_distance):
        """Sets the max_distance of this V3SearchParameters.

        Filter by maximum distance (in metres) from location specified via latitude and longitude parameters  # noqa: E501

        :param max_distance: The max_distance of this V3SearchParameters.  # noqa: E501
        :type: float
        """

        self._max_distance = max_distance

    @property
    def include_addresses(self):
        """Gets the include_addresses of this V3SearchParameters.  # noqa: E501

        Placeholder for future development; currently unavailable  # noqa: E501

        :return: The include_addresses of this V3SearchParameters.  # noqa: E501
        :rtype: bool
        """
        return self._include_addresses

    @include_addresses.setter
    def include_addresses(self, include_addresses):
        """Sets the include_addresses of this V3SearchParameters.

        Placeholder for future development; currently unavailable  # noqa: E501

        :param include_addresses: The include_addresses of this V3SearchParameters.  # noqa: E501
        :type: bool
        """

        self._include_addresses = include_addresses

    @property
    def include_outlets(self):
        """Gets the include_outlets of this V3SearchParameters.  # noqa: E501

        Indicates if outlets will be returned in response (default = true)  # noqa: E501

        :return: The include_outlets of this V3SearchParameters.  # noqa: E501
        :rtype: bool
        """
        return self._include_outlets

    @include_outlets.setter
    def include_outlets(self, include_outlets):
        """Sets the include_outlets of this V3SearchParameters.

        Indicates if outlets will be returned in response (default = true)  # noqa: E501

        :param include_outlets: The include_outlets of this V3SearchParameters.  # noqa: E501
        :type: bool
        """

        self._include_outlets = include_outlets

    @property
    def match_stop_by_suburb(self):
        """Gets the match_stop_by_suburb of this V3SearchParameters.  # noqa: E501

        Indicates whether to find stops by suburbs in the search term (default = true)  # noqa: E501

        :return: The match_stop_by_suburb of this V3SearchParameters.  # noqa: E501
        :rtype: bool
        """
        return self._match_stop_by_suburb

    @match_stop_by_suburb.setter
    def match_stop_by_suburb(self, match_stop_by_suburb):
        """Sets the match_stop_by_suburb of this V3SearchParameters.

        Indicates whether to find stops by suburbs in the search term (default = true)  # noqa: E501

        :param match_stop_by_suburb: The match_stop_by_suburb of this V3SearchParameters.  # noqa: E501
        :type: bool
        """

        self._match_stop_by_suburb = match_stop_by_suburb

    @property
    def match_route_by_suburb(self):
        """Gets the match_route_by_suburb of this V3SearchParameters.  # noqa: E501

        Indicates whether to find routes by suburbs in the search term (default = true)  # noqa: E501

        :return: The match_route_by_suburb of this V3SearchParameters.  # noqa: E501
        :rtype: bool
        """
        return self._match_route_by_suburb

    @match_route_by_suburb.setter
    def match_route_by_suburb(self, match_route_by_suburb):
        """Sets the match_route_by_suburb of this V3SearchParameters.

        Indicates whether to find routes by suburbs in the search term (default = true)  # noqa: E501

        :param match_route_by_suburb: The match_route_by_suburb of this V3SearchParameters.  # noqa: E501
        :type: bool
        """

        self._match_route_by_suburb = match_route_by_suburb

    @property
    def match_stop_by_gtfs_stop_id(self):
        """Gets the match_stop_by_gtfs_stop_id of this V3SearchParameters.  # noqa: E501

        Indicates whether to search for stops according to a metlink stop ID (default = false)  # noqa: E501

        :return: The match_stop_by_gtfs_stop_id of this V3SearchParameters.  # noqa: E501
        :rtype: bool
        """
        return self._match_stop_by_gtfs_stop_id

    @match_stop_by_gtfs_stop_id.setter
    def match_stop_by_gtfs_stop_id(self, match_stop_by_gtfs_stop_id):
        """Sets the match_stop_by_gtfs_stop_id of this V3SearchParameters.

        Indicates whether to search for stops according to a metlink stop ID (default = false)  # noqa: E501

        :param match_stop_by_gtfs_stop_id: The match_stop_by_gtfs_stop_id of this V3SearchParameters.  # noqa: E501
        :type: bool
        """

        self._match_stop_by_gtfs_stop_id = match_stop_by_gtfs_stop_id

    def to_dict(self):
        """Returns the model properties as a dict"""
        result = {}

        for attr, _ in six.iteritems(self.swagger_types):
            value = getattr(self, attr)
            if isinstance(value, list):
                result[attr] = list(map(
                    lambda x: x.to_dict() if hasattr(x, "to_dict") else x,
                    value
                ))
            elif hasattr(value, "to_dict"):
                result[attr] = value.to_dict()
            elif isinstance(value, dict):
                result[attr] = dict(map(
                    lambda item: (item[0], item[1].to_dict())
                    if hasattr(item[1], "to_dict") else item,
                    value.items()
                ))
            else:
                result[attr] = value
        if issubclass(V3SearchParameters, dict):
            for key, value in self.items():
                result[key] = value

        return result

    def to_str(self):
        """Returns the string representation of the model"""
        return pprint.pformat(self.to_dict())

    def __repr__(self):
        """For `print` and `pprint`"""
        return self.to_str()

    def __eq__(self, other):
        """Returns true if both objects are equal"""
        if not isinstance(other, V3SearchParameters):
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        """Returns true if both objects are not equal"""
        return not self == other